keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -2025,6 +2006,152 @@ Beta:
+  If the `apiserver_flowcontrol_dispatched_requests_total` metric is non-zero,
+  this feature is in use. Note that this isn't a workload feature, but a
+  control plane one.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [x] Metrics
+    - Metric name: `apiserver_flowcontrol_request_queue_length_after_enqueue`
+    - Components exposing the metric: kube-apiserver
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  No SLOs are proposed for the above SLI.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?** No.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+### Scalability
+
+<!-- I don't have the knowledge to answer some questions in this section. -->
+
+* **Will enabling / using this feature result in any new API calls?** No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Yes, a new flowcontrol API group, configuration types, and status types are
+  introduced. See `staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go` for a
+  full list.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+  provider?** No.
+
+* **Will enabling / using this feature result in increasing size or count of
+  the existing API objects?** No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs]?** Yes, a non-negligible latency
+  is added to API calls to kube-apiserver. While [preliminary tests](https://github.com/tkashem/graceful/blob/master/priority-fairness/filter-latency/readme.md)
+  shows that the API server latency is still well within the existing SLOs,
+  more thorough testing needs to be performed.
+
+* **Will enabling / using this feature result in non-negligible increase of
+  resource usage (CPU, RAM, disk, IO, ...) in any components?** The proposed
+  flowcontrol logic in request handling in kube-apiserver will increase the CPU
+  and memory overheads involved in serving each request.
+
+### Troubleshooting
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  The feature is itself within the API server. Etcd being unavailable would
+  likely cause kube-apiserver to fail at processing incoming requests.
+
+* **What are other known failure modes?** A misconfiguration could reject
+  requests incorrectly. See the rollout and monitoring sections for details on
+  which metrics to watch to detect such a failure (see the `kep.yaml` file for
+  the full list of metrics). The following kube-apiserver log messages could
+  also indicate potential issues:
+  - "Unable to list PriorityLevelConfiguration objects"
+  - "Unable to list FlowSchema objects"
+  <!-- Should there be a higher V error message when requests are rejected? -->
```

Comment:
Add: We believe that it's impossible for the cluster to be configured into a state where the cluster administrator can't fix it by sending requests to edit flow control / priority level objects.

Reply:
Not solved yet.

Reply:
See paragraph above.